src/org/zaproxy/zap/extension/autoupdate/AddOnsTableModel.java

```diff
@@ -110,9 +122,11 @@ protected void addAddOnWrapper(AddOn addOn, AddOnWrapper.Status status) {
         AddOnWrapper aow = createAddOnWrapper(addOn, status);
         int idx = 0;
         for (; idx < getAddOnWrappers().size(); idx++) {
-            if (comparator.compare(aow, getAddOnWrappers().get(idx)) < 0) {
-                break;
-            }
+        	if (comparator != null) {
```

Comment:
Consider moving this if statement to outside the loop, no need to loop if there's no comparator (and in that case the idx is just the last position, i.e. getAddOnWrappers().size().)